docs/cupynumeric/source/user/Legate Task tutorial.rst

```diff
@@ -0,0 +1,954 @@
+       z_local = xp.asarray(z)
+       z_local[:] = a * x_local + y_local
+
+    
+    parser = argparse.ArgumentParser(description="Run SAXPY operation.")
+    parser.add_argument("--size", type=int, default=1000, help="Size of input arrays")
+    args = parser.parse_args()
+    size = args.size
+    
+    x_global = cpn.arange(size, dtype=cpn.float32)
+    y_global = cpn.ones(size, dtype=cpn.float32)
+    z_global = cpn.zeros(size, dtype=cpn.float32)
+        
+    rt = lg.get_legate_runtime()
+    
+    #warm-up run
+    saxpy_task(x_global, y_global, z_global, 2.0)
+    
+    rt.issue_execution_fence()
+    start = time()
+    saxpy_task(x_global, y_global, z_global, 2.0)
+    rt.issue_execution_fence()
+    end = time()
+    
+    print(f"\nTime elapsed for saxpy: {(end - start)/1000:.6f} milliseconds")
+
+.. _section-2:
+
+Running on CPU and GPU - `Guide <https://docs.nvidia.com/legate/latest/usage.html>`__
+-------------------------------------------------------------------------------------
+
+In order to run the program, use the legate launcher, and include any
+flags necessary like --cpus, --gpus, and more. If you want to run
+specifically only on CPU, you must include the flag “--gpus 0”.
+
+The Legate runtime is used in the main function to control and
+synchronize task execution. The get_legate_runtime() function returns
+this runtime, which is used to issue commands like execution fences. In
+this example, issue_execution_fence() is called before and after the
+saxpy_task to ensure accurate time measurement. Since Legate tasks run
+asynchronously by default, these fences make the program wait until all
+previous tasks have finished, so the measured time reflects only the
+actual task execution. This is a common pattern when precise timing,
+synchronization, or ordered execution of asynchronous tasks is needed.
+
+GPU initialization has a fixed setup time that can significantly affect
+the runtime when processing small arrays. Using a large input (e.g., 100
+million elements) ensures that the computation time outweighs the
+startup overhead, giving more realistic timing results. Since the first
+GPU run may include the setup overhead like compilation or memory
+allocation, a warm-up pass helps eliminate these one-time costs from
+performance measurements, ensuring more reliable results.
+
+Let’s set the input array size to 100 million elements to better
+evaluate the speedup from distributed computing with GPUs.
+
+**CPU** - To run with CPU, use the following command.
+
+.. code-block:: sh
+
+    legate --cpus 1 --gpus 0 ./saxpy.py --size 100000000
+
+Output:
+
+.. code-block:: text
+
+    Time elapsed for saxpy: 146.303000 milliseconds
+
+**GPU** - To run with GPU, use the following command.
+
+.. code-block:: sh
+
+    legate --gpus 2 ./saxpy.py --size 100000000
+
+Output:
+
+.. code-block:: text
+
+    Time elapsed for saxpy : 1.949000 milliseconds
+
+Multi-Node: Refer to the Legate documentation on how to run
```

Comment:
"Multi-Node:" should get the same treatment as CPU and GPU above for consistency (i.e. actually be a sub-sub section, or a rubric)